src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -141927,5 +141927,376 @@ static T GetValue2<T>(Collection c, object key)
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I<(object a, object b)>, I2<(object notA, object notB)> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (9,11): error CS8140: 'I<(object notA, object notB)>' is already listed in the interface list on type 'I3' with different tuple element names, as 'I<(object a, object b)>'.
+                // interface I3 : I<(object a, object b)>, I2<(object notA, object notB)> { }
+                Diagnostic(ErrorCode.ERR_DuplicateInterfaceWithTupleNamesInBaseList, "I3").WithArguments("I<(object notA, object notB)>", "I<(object a, object b)>", "I3").WithLocation(9, 11),
+                // (16,15): error CS0229: Ambiguity between 'I<(object a, object b)>.Item' and 'I<(object notA, object notB)>.Item'
+                //         _ = i.Item;
+                Diagnostic(ErrorCode.ERR_AmbigMember, "Item").WithArguments("I<(object a, object b)>.Item", "I<(object notA, object notB)>.Item").WithLocation(16, 15)
+                );
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I<(object a, object b)>", "I2<(object notA, object notB)>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I<(object a, object b)>", "I2<(object notA, object notB)>", "I<(object notA, object notB)>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+        }
+
+        [Fact]
+        public void AmbigMember_NoDifference()
+        {
+            var src = @"
+interface I<T> { T Item { get; } }
+
+interface I2<T> : I<T> { }
+
+interface I3 : I<object>, I2<object> { }
+
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                );
+            var i3 = comp.GetTypeByMetadataName("I3");
+            AssertEx.Equal(new string[] { "I<object>", "I2<object>" },
+                i3.Interfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+            AssertEx.Equal(new string[] { "I2<object>", "I<object>" },
+                i3.AllInterfaces().ToTestDisplayStrings(TypeWithAnnotations.TestDisplayFormat));
+        }
+
+        [Fact]
+        public void AmbigMember_DifferenceBetweenNonnullableAndOblivious_WithoutConstraint()
+        {
+            var src = @"
+#nullable disable
+interface I<T> { T Item { get; set; } }
+
+#nullable enable
+interface I2<T> : I<T> { }
+
+#nullable disable
+interface I3 : I<object>, I2<object> { }
+
+#nullable enable
+public class C
+{
+    void M(I3 i)
+    {
+        _ = i.Item;
+        i.Item = null;
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
```

Comment:
>comp.VerifyDiagnostics(); [](start = 12, length = 25)

In this and other tests that are supposed to verify that an ambiguity is successfully resolved, I think we should verify how exactly it is resolved to ensure that the resolution is deterministic, etc.

Reply:
We're already verifying this in two ways: the behavior of `.Item` during nullability analysis, and by checking how the ambiguity is resolved in `AllInterfaces`.
I could also check `GetSymbolInfo` on `i.Item`, but that doesn't seem to add much. Is that what you had in mind?

---
In reply to: [524699086](https://github.com/dotnet/roslyn/pull/49338#discussion_r524699086) [](ancestors = 524699086)

Reply:
> and by checking how the ambiguity is resolved in ```AllInterfaces```

As far as I can tell, there is no change to the behavior of this API. What could we be checking?

> I could also check GetSymbolInfo on i.Item, but that doesn't seem to add much. Is that what you had in mind?

Yes. something like that, I think we need to assert specific symbol. We probably should also test lookup APIs that SemanticModel offers.

---
In reply to: [542852918](https://github.com/dotnet/roslyn/pull/49338#discussion_r542852918) [](ancestors = 542852918,524699086)